docs/reference/glossary.md

```diff
@@ -0,0 +1,113 @@
+
+## C
+
+## D
+
+**Data Availability**
+
+Data availability is the guarantee that the block proposers are required to publish the data for each block, which nodes participating in Ethereum's consensus store locally. This is what powers trustlessness: rather than having to trust that block producers are honest, all nodes on the Ethereum network execute the transactions to verify that the new information is valid. Data availability is one of the ways in which a layer 2 differs from a sidechain.
+
+## E
+
+**Ethereum Virtual Machine (EVM)**
+
+A stack-based virtual machine that executes bytecode. In Ethereum, the execution model specifies how the system state is altered given a series of bytecode instructions and a small tuple of environmental data. This is specified through a formal model of a virtual state machine.
+
+## F
+
+## G
+
+## H
+
+## I
+
+## J
+
+## K
+
+## L
+
+**Layer 1**
+
+Layer 1 (L1) is the underlying foundation and base blockchain that various layer 2 (L2) networks build on top of. For example, Ethereum is an L1 that is comprised of node operators to secure and validate the network, block producers, the history of transaction data, and the consensus mechanism of the blockchain itself.
+
+**Layer 2**
+
+L2 is a separate blockchain that increases transaction speed and throughput, while fully or partially deriving its security from Ethereum. Additionally, Layer 2 projects must rely on Ethereum for data availability by posting their transaction data onto Ethereum. L2s decrease L1 congestion by bundling transactions to be submitted to Ethereum, thus increasing scalability while inheriting Ethereum’s data availability, security, and decentralization.
+
+## M
+
+## N
+
+## O
+
+**Optimistic Rollup**
+
+A rollup of transactions that use fraud proofs to offer increased layer 2 transaction throughput while using the security provided by Mainnet (layer 1). Unlike Plasma, a similar layer 2 solution, Optimistic rollups can handle more complex transaction types – anything possible in the EVM. However, in order to release assets back on the layer 1, users must wait for a 7 day challenge period. Liquidity providers can help alleviate distributing L1 ETH immediately, but this relies on trust in the security of the provider, and does not provide a mechanism for transferring NFTs.
+
+## P
+
+## Q
+
+## R
+
+**Rollup**
+
+A type of layer 2 scaling solution that batches multiple transactions and submits them to the Ethereum main chain in a single transaction. This allows for reductions in gas costs and increases in transaction throughput. There are Optimistic and Zero-knowledge rollups which use different security methods to offer these scalability gains.
```

Comment:
```suggestion
A type of layer 2 scaling solution that batches multiple transactions and submits them to the Ethereum main chain in a single transaction. This allows for reductions in gas costs and increases in transaction throughput. There are optimistic and zero-knowledge rollups which use different security methods to offer these scalability gains.
```

Reply:
We can possible also link optimistic and zero-knowledge rollups to the sections in this glossary